submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/submit/async/SubmitAsyncResource.kt

```diff
@@ -0,0 +1,98 @@
+package ac.uk.ebi.biostd.submission.web.resources.submit.async
+
+import ac.uk.ebi.biostd.integration.SubFormat.Companion.JSON_PRETTY
+import ac.uk.ebi.biostd.integration.SubFormat.Companion.TSV
+import ac.uk.ebi.biostd.integration.SubFormat.Companion.XML
+import ac.uk.ebi.biostd.submission.converters.BioUser
+import ac.uk.ebi.biostd.submission.web.handlers.SubmitWebHandler
+import ac.uk.ebi.biostd.submission.web.model.ContentSubmitWebRequest
+import ac.uk.ebi.biostd.submission.web.model.OnBehalfRequest
+import ebi.ac.uk.extended.model.FileMode
+import ebi.ac.uk.model.constants.APPLICATION_JSON
+import ebi.ac.uk.model.constants.ATTRIBUTES
+import ebi.ac.uk.model.constants.FILE_MODE
+import ebi.ac.uk.model.constants.SUBMISSION_TYPE
+import ebi.ac.uk.model.constants.TEXT_PLAIN
+import ebi.ac.uk.model.constants.TEXT_XML
+import ebi.ac.uk.security.integration.model.api.SecurityUser
+import org.springframework.http.MediaType.APPLICATION_JSON_VALUE
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.ResponseBody
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/submissions/async")
+@PreAuthorize("isAuthenticated()")
+@Suppress("LongParameterList")
+class SubmitAsyncResource(private val submitWebHandler: SubmitWebHandler) {
```

Comment:
It would be nice to add the corresponding tests using mock mvc

Reply:
I will add thos ein next pr to solve issue right now